.github/workflows/release.yml

```diff
@@ -14,18 +14,75 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v3
+
     - name: Load Secrets from Vault
       uses: rancher-eio/read-vault-secrets@main
       with:
         secrets: |
           secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials username | DOCKER_USERNAME ;
           secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials password | DOCKER_PASSWORD ;
 
-    - name: Checkout Repo
-      uses: actions/checkout@v3
-
     - name: Login to Docker Hub
       uses: docker/login-action@v3
       with:
         username: ${{ env.DOCKER_USERNAME }}
         password: ${{ env.DOCKER_PASSWORD }}
+
+    # Fail for non-tagged commits
+    - name: Tag Check
```

Comment:
If I understand correctly, this job only triggers on pushing a tag. Do we need this step?

Reply:
The job could be also triggered manually with the `workflow_dispatch`, so this check was added to be sure that the build is running against an existing tag.

We can remove it if we want to push "dev" version of the cli.

Reply:
why do we want to trigger a release manually? In my opinion a release should always be triggered by a tag. I'd remove this tag check and `workflow_dispatch`

Reply:
No, but in the past I found convenient to have a way to manually trigger a workflow.

Thinking a bit more about this you still have the chance to re-trigger a failed build, so maybe it's not needed. I'll remove it. :)

Reply:
Done eb27b85